### PR TITLE
StateProof recoverability CR fixes

### PR DESCRIPTION
**Pull request by algoidan**

### CODE REVIEW

File: ledger/stateproofverificationtracker.go [hunk 1]
```diff
@@ -169,41 +167,27 @@ func (spt *stateProofVerificationTracker) close() {
...
 	}
 
-	verificationContext, err := spt.lookUpVerificationContext(stateProofLastAttestedRound)
+	verificationContext, err := spt.lookupVerificationContext(stateProofLastAttestedRound)
 	if err != nil {
 		return nil, err
 	}
 
-	spt.mu.Lock()
 	spt.lastLookedUpVerificationContext = *verificationContext
```
Comment (algorandskiy) [inline ledger/stateproofverificationtracker.go:184]: we taking RLock above but have write op here

**Review by algorandskiy — COMMENTED** [review]
Looks good - only one rlock vs lock finding below

File: ledger/stateproofverificationtracker.go (hunk 1 above)
Comment (algoidan) [reply ledger/stateproofverificationtracker.go:184]: now I'm not sure I like this change. This change will cause the lookup function to take a write lock the entire time. Before it only took it when updating the cache 
What do you think?

File: ledger/stateproofverificationtracker.go [hunk 2]
```diff
@@ -168,41 +166,27 @@ func (spt *stateProofVerificationTracker) close() {
 }
 
 func (spt *stateProofVerificationTracker) LookupVerificationContext(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationContext, error) {
-	if lstlookup := spt.retrieveFromCache(stateProofLastAttestedRound); lstlookup != nil {
-		return lstlookup, nil
+	spt.mu.RLock()
```
Comment (algorandskiy) [inline ledger/stateproofverificationtracker.go:169]:
```suggestion
	defer func() {
		spt.mu.Lock()
		spt.lastLookedUpVerificationContext = *verificationContext
		spt.mu.Unlock()
	}
	spt.mu.RLock()
```

**Pull request merged**